vega/base.py

```diff
@@ -18,8 +19,8 @@ class VegaBase(object):
 
     def __init__(self, spec, data=None):
         """Initialize the visualization object."""
-        spec = self._prepare_spec(spec, data)
-        self.spec = utils.update(spec, self.DEFAULTS, overwrite=False)
+        spec = self._prepare_spec(copy.deepcopy(spec), data)
+        self.spec = utils.update(spec, self.DEFAULTS)
```

Comment:
The problem with calling deepcopy here is that if a pandas dataframe is in the spec, it will make a copy of this as well. I think instead we want something like ``self._prepare_spec(utils.update({}, spec), data)``